core-bundle/tests/DependencyInjection/Compiler/ConfigureFilesystemPassTest.php

```diff
@@ -57,4 +72,86 @@ function (FilesystemConfiguration $config) use ($container): bool {
 
         (new ConfigureFilesystemPass())->process($container);
     }
+
+    /**
+     * @dataProvider provideSymlinks
+     */
+    public function testCreatesMountsForSymlinks(string $target, string $link): void
+    {
+        $tempDir = $this->getTempDir();
+        $target = str_replace('<root>', $tempDir, $target);
+
+        // Setup directories with symlink
+        $filesystem = new Filesystem();
+        $filesystem->mkdir(Path::join($tempDir, 'files'));
+        $filesystem->dumpFile(Path::join($tempDir, 'vendor/foo/dummy.txt'), 'dummy');
+
+        $this->createSymlink($target, $link, Path::join($tempDir, 'files'));
+
+        $container = new ContainerBuilder(
+            new ParameterBag([
+                'kernel.project_dir' => $tempDir,
+                'contao.upload_path' => 'files',
+            ])
+        );
+
+        $container->setDefinition(
+            $mountManagerId = 'contao.filesystem.mount_manager',
+            $mountManagerDefinition = new Definition(MountManager::class)
+        )->setPublic(true);
+
+        (new ConfigureFilesystemPass())->process($container);
+
+        $methodCalls = $mountManagerDefinition->getMethodCalls();
+
+        $this->assertCount(1, $methodCalls);
+        $this->assertSame('mount', $methodCalls[0][0]);
+
+        [$reference, $mountPath] = $methodCalls[0][1];
+        $this->assertInstanceOf(Reference::class, $reference);
+        $this->assertSame('files/foo', Path::normalize($mountPath));
+
+        $adapter = $container->get((string) $reference);
+        $this->assertInstanceOf(LocalFilesystemAdapter::class, $adapter);
+        $this->assertSame('dummy', $adapter->read('dummy.txt'));
+
+        $container->compile();
+
+        $mountManager = $container->get($mountManagerId);
+        $this->assertSame('dummy', $mountManager->read('files/foo/dummy.txt'));
+
+        // Cleanup
+        $filesystem->remove(Path::join($tempDir, 'files'));
+        $filesystem->remove(Path::join($tempDir, 'vendor'));
+    }
+
+    public function provideSymlinks(): \Generator
+    {
+        yield 'absolute symlink' => [
+            '<root>/vendor/foo',
+            'foo',
+        ];
+
+        yield 'symlink relative to the files directory' => [
+            '../vendor/foo',
+            'foo',
+        ];
+    }
+
+    private function createSymlink(string $target, string $link, string $cwd): void
+    {
+        if ('\\' === \DIRECTORY_SEPARATOR) {
+            $target = str_replace('/', '\\', $target);
+            $link = str_replace('/', '\\', $link);
+
+            Process::fromShellCommandline('mklink /d "${:link}" "${:target}"', $cwd)
+                ->mustRun(null, ['link' => $link, 'target' => $target])
+            ;
+        } else {
+            chdir($cwd);
```

Comment:
This should get reset in the tearDown() method 🤓

Reply:
Should we record the current path with `getcwd` or is there a default we could use?

Reply:
I think recording is the way to go…

Reply:
implemented in a626f31